src/convertToObject.js

```diff
@@ -6,7 +6,32 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const mainObj = {};
+
+  const workArr = sourceString.split(';');
+
+  const cleanArr = workArr
+    .map(function (item) {
+      const result = item.replace(/\n/g, '').trim();
+
+      return result;
+    })
+    .filter((item) => item !== '');
+
+  for (const pair of cleanArr) {
+    const index = pair.indexOf(':');
+
+    if (index > -1) {
+      const key = pair.slice(0, index);
```

Comment:
The variable `key` is only used as an intermediate step before trimming. You can combine this into one line to improve readability, for example: `const key = pair.slice(0, index).trim();`